sopel/builtins/xkcd.py

```diff
@@ -124,13 +118,16 @@ def xkcd(bot, trigger):
             if (query.lower() == "latest" or query.lower() == "newest"):
                 requested = latest
             else:
-                number = searchxkcd_search(query)
-                if number is None:
-                    # generic web-search engine as fallback
-                    number = web_search(query)
-                if not number:
-                    bot.reply('Could not find any comics for that query.')
+                try:
+                    number = searchxkcd_search(query)
+                except NoResultsError:
+                    bot.reply("Sorry, I couldn't find any comics for that query.")
                     return
+                except (ResponseFormatError, SearchFailedError, SearchXkcdError):
+                    bot.reply(
+                        "A technical problem prevented me from searching. "
+                        "Please ask my owner to check my logs.")
```

Comment:
```suggestion
                        "Please ask my owner to check my logs.")
                        return
```

Don't want to proceed without a search result, do we?